Source/santactl/Commands/SNTCommandMonitorMode.mm

```diff
@@ -0,0 +1,121 @@
+  return (@"Usage: santactl monitormode [options]\n"
+          @"  Options:\n"
+          @"    --duration {minutes}: An optional number of minutes of temporary Monitor Mode\n"
+          @"                          to request. By default, will use configured time allotted\n"
+          @"                          by policy.\n"
+          @"    --cancel: End temporary Monitor Mode and revert to Lockdown Mode.\n"
+          @"\n");
+}
+
++ (NSSet<NSString *> *)aliases {
+  return [NSSet setWithArray:@[ @"mm" ]];
+}
+
+- (void)runWithArguments:(NSArray *)arguments {
+  NSNumber *requestedDuration;
+  bool shouldCancel = false;
+
+  // Parse arguments
+  for (NSUInteger i = 0; i < arguments.count; ++i) {
+    NSString *arg = arguments[i];
+
+    if ([arg caseInsensitiveCompare:@"--duration"] == NSOrderedSame) {
+      if (++i > arguments.count - 1) {
+        [self printErrorUsageAndExit:@"--duration requires an argument"];
+      }
+
+      arg = arguments[i];
+      if (arg.length == 0) {
+        [self printErrorUsageAndExit:@"--duration requires a whole number argument"];
+      }
+
+      NSCharacterSet *nonDigitCharacterSet =
+          [[NSCharacterSet decimalDigitCharacterSet] invertedSet];
+      if ([arg rangeOfCharacterFromSet:nonDigitCharacterSet].location != NSNotFound) {
+        [self printErrorUsageAndExit:@"--duration requires a whole number argument"];
+      }
+
+      NSInteger val = [arg integerValue];
+      if (val <= 0) {
+        [self printErrorUsageAndExit:@"--duration must be greater than 0"];
+      }
+
+      requestedDuration = @(val);
+    } else if ([arg caseInsensitiveCompare:@"--cancel"] == NSOrderedSame) {
+      shouldCancel = true;
+    }
+  }
+
+  __block BOOL success;
+
+  if (shouldCancel) {
+    [[self.daemonConn synchronousRemoteObjectProxy] cancelTemporaryMonitorMode:^(NSError *err) {
+      success = (err == nil);
+      if (err) {
+        TEE_LOGE(@"Unable cancel Monitor Mode: %@", err.localizedDescription);
+        return;
+      }
+    }];
+  } else {
+    [[self.daemonConn synchronousRemoteObjectProxy]
+        requestTemporaryMonitorModeWithDuration:requestedDuration
+                                          reply:^(uint32_t minutes, NSError *err) {
+                                            success = (err == nil);
+                                            if (err) {
+                                              TEE_LOGE(@"Unable to enter Monitor Mode: %@",
+                                                       err.localizedDescription);
+                                              return;
+                                            }
+
+                                            TEE_LOGI(
+                                                @"Temporarily switching to Monitor Mode for %u %@",
+                                                minutes, minutes > 1 ? @"minutes" : @"minute");
+                                          }];
+  }
+
+  exit(success ? EXIT_SUCCESS : EXIT_FAILURE);
```

Comment:
I don't know if it's _guaranteed_ that the reply block has finished executing when a method call on a synchronous remote object proxy returns. It might be safer to move the exit() call into the reply block to ensure the log lines are always printed.